src/Evolution/Systems/NewtonianEuler/NumericalFluxes/Hllc.cpp

```diff
@@ -0,0 +1,212 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/NewtonianEuler/NumericalFluxes/Hllc.hpp"
+
+#include <algorithm>
+#include <array>
+#include <limits>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/FaceNormal.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Evolution/Systems/NewtonianEuler/Characteristics.hpp"
+#include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+/// \cond
+namespace NewtonianEuler {
+namespace NumericalFluxes {
+
+template <size_t Dim, typename Frame>
+void Hllc<Dim, Frame>::package_data(
+    const gsl::not_null<Variables<package_tags>*> packaged_data,
+    const Scalar<DataVector>& normal_dot_flux_mass_density,
+    const tnsr::I<DataVector, Dim, Frame>& normal_dot_flux_momentum_density,
+    const Scalar<DataVector>& normal_dot_flux_energy_density,
+    const Scalar<DataVector>& mass_density,
+    const tnsr::I<DataVector, Dim, Frame>& momentum_density,
+    const Scalar<DataVector>& energy_density,
+    const tnsr::I<DataVector, Dim, Frame>& velocity,
+    const Scalar<DataVector>& pressure,
+    const db::const_item_type<char_speeds_tag> characteristic_speeds,
+    const tnsr::i<DataVector, Dim, Frame>& interface_unit_normal) const
+    noexcept {
+  get<::Tags::NormalDotFlux<Tags::MassDensityCons<DataVector>>>(
+      *packaged_data) = normal_dot_flux_mass_density;
+  get<::Tags::NormalDotFlux<Tags::MomentumDensity<DataVector, Dim, Frame>>>(
+      *packaged_data) = normal_dot_flux_momentum_density;
+  get<::Tags::NormalDotFlux<Tags::EnergyDensity<DataVector>>>(*packaged_data) =
+      normal_dot_flux_energy_density;
+  get<Tags::MassDensityCons<DataVector>>(*packaged_data) = mass_density;
+  get<Tags::MomentumDensity<DataVector, Dim, Frame>>(*packaged_data) =
+      momentum_density;
+  get<Tags::EnergyDensity<DataVector>>(*packaged_data) = energy_density;
+  get<Tags::Pressure<DataVector>>(*packaged_data) = pressure;
+  get<::Tags::Normalized<::Tags::UnnormalizedFaceNormal<Dim, Frame>>>(
+      *packaged_data) = interface_unit_normal;
+  get<NormalVelocity>(*packaged_data) =
+      dot_product(interface_unit_normal, velocity);
+
+  // When packaging interior data, FirstSpeedEstimate and SecondSpeedEstimate
+  // will hold the min and max char speeds, respectively. On the other hand,
+  // when packaging exterior data, the characteristic speeds will be computed
+  // along *minus* the exterior normal, so FirstSpeedEstimate will hold *minus*
+  // the max speed, while SecondSpeedEstimate will store *minus* the min speed.
```

Comment:
For my understanding: are you making the distinction between interior/exterior sides of an interface or mortar between two elements? or are you making the distinction between internal and external boundaries of the computational domain?

Reply:
Here I mean interior and exterior sides of a mortar between two elements. Sorry for the confusion, I used this nomenclature in order to make it consistent with other numerical fluxes in spectre. I figured that keeping consistency would make things easier to change when we rework the flux comm or the num fluxes.